File: Python/566. Reshape the Matrix.py
# Time: O(r*c)
class Solution:
    def matrixReshape(self, nums, r, c):
        """
        :type nums: List[List[int]]
        :type r: int
        :type c: int
        :rtype: List[List[int]]
        """
        R = len(nums)
        C = len(nums[0])
        ans = list()
        if R*C == r*c:
            tmp = list()
            cnt = 0
            for i in range(R):
                for j in range(C):
                    tmp.append(nums[i][j])
                    cnt += 1
                    if cnt == c:
                        cnt = 0
                        ans.append(tmp)
                        tmp = list()
            return ans
        else:
            return nums

File: C++/566. Reshape the Matrix.cpp
// Time: O(r*c)
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& nums, int r, int c) {
        int R = nums.size(), C = nums[0].size();
        if(R*C != r*c)
            return nums;
        vector<vector<int>> ans;
        vector<int> tmp;
        int cnt = 0;
        for(int i=0;i<R;i++)
        {
            for(int j=0;j<C;j++)
            {
                tmp.push_back(nums[i][j]);
                cnt++;
                if(cnt==c)
                {
                    ans.push_back(tmp);
                    cnt = 0;
                    tmp.clear();
                }
            }
        }
        return ans;
    }
};